### PR TITLE
NAS-116484 / 22.02.2 / optimize disk.sync_all on SCALE

### PR DESCRIPTION
**Pull request by yocalebo**
Same principles that were done in 13 here https://github.com/truenas/middleware/pull/8937 are applied here to SCALE.

Differences in time that I've measured:

Inserting 1255 _new_ disk entries into db OLD took 1:54.17
Inserting 1255 _new_ disk entries into db NEW takes 1:00.24 **(~62% faster)**

Updating 1255 _existing_ disk entries into db OLD took 20:04.96
Updating 1255 _existing_ disk entries into db NEW takes 1:20.05 **(~175% faster)**

We're still missing some optimizations on SCALE (like `enclosure.sync_disk`) but this is a good first step.

### CODE REVIEW

**Comment by yocalebo** [comment]
retest this please

**Comment by yocalebo** [comment]
ping @truenas/middleware . This has been tested on real hardware and is essentially a SCALE version of what we're already doing in CORE.

**Comment by rick-mesta** [comment]
@themylogin / @sonicaj / @anodos325 / @Qubad786: Gents, please make time to provide timely feedback on `22.02.2` PR's. Freeze date is almost upon us -- 06/08/2022

File: src/middlewared/middlewared/plugins/device_/device_info_base.py [hunk 1]
```diff
@@ -43,7 +43,7 @@ async def get_serials(self):
         raise NotImplementedError()
 
     @private
```
Comment (themylogin) [inline src/middlewared/middlewared/plugins/device_/device_info_base.py:45]: We don't need this files separation anymore, can you please merge them two into just `device_info.py`?

File: src/middlewared/middlewared/plugins/disk_/sync.py [hunk 2]
```diff
@@ -79,116 +73,189 @@ async def sync_all(self, job):
         else:
             self.logger.info('Found %d disks', number_of_disks)
 
+        return number_of_disks
+
+    @private
```
Comment (themylogin) [inline src/middlewared/middlewared/plugins/disk_/sync.py:78]: Both methods are great candidates for unit tests, for the future team members to have a list of examples of how we build, parse and map disk identifiers.

**Review by themylogin — COMMENTED** [review]
The logic looks great, much cleaner now.

File: src/middlewared/middlewared/plugins/device_/device_info_base.py (hunk 1 above)
Comment (yocalebo) [reply src/middlewared/middlewared/plugins/device_/device_info_base.py:45]: Yep, no problem will do

File: src/middlewared/middlewared/plugins/disk_/sync.py (hunk 2 above)
Comment (yocalebo) [reply src/middlewared/middlewared/plugins/disk_/sync.py:78]: Sure, no problem. You're talking about `dev_to_ident` and `ident_to_dev` right?

Comment (themylogin) [reply src/middlewared/middlewared/plugins/disk_/sync.py:78]: Yes

**Comment by yocalebo** [comment]
@themylogin think I've got everything that you asked for

**Review by themylogin — APPROVED** [review]
LGTM, excellent job, thank you! 👍

**Pull request merged**